NetKet/Machine/hypercube_conv_layer.hpp

```diff
@@ -35,33 +35,36 @@ namespace netket {
  be column major.
  */
 template <typename T>
-class Convolutional : public AbstractLayer<T> {
+class ConvolutionalHypercube : public AbstractLayer<T> {
   using VectorType = typename AbstractLayer<T>::VectorType;
   using MatrixType = typename AbstractLayer<T>::MatrixType;
   using VectorRefType = typename AbstractLayer<T>::VectorRefType;
   using VectorConstRefType = typename AbstractLayer<T>::VectorConstRefType;
 
   static_assert(!MatrixType::IsRowMajor, "MatrixType must be column-major");
 
-  std::shared_ptr<const AbstractGraph> graph_;
```

Comment:
I think it's a bit risky if we don't store any graph object in here. 
This Layer should just take a Hypercube object and use its methods, not do everything from scratch! Are there methods missing in Hypercube that you need and can't use yet?

Reply:
Hmm but in principle here if use a custom graph, e.g. the j1j2 we can still use this conv net.

Reply:
And put stride as well which changes the graph

Reply:
I agree, but we can't have a machine completely disconneted from the graph used for the simulation! We still use the graph for sampling/defining operators etc... If there are inconsistencies between the graph used in the machine and the one used for all the other tasks, it's going to be a problem! 
What are the methods you need from the graph?

Reply:
Well but the graph for each layer is not the same already if u have stride

Reply:
So i need to have a new graph for each layer?